build/root/Makefile.generated_files

```diff
@@ -405,7 +405,7 @@ OPENAPI_OUTPUT_PKG := pkg/generated/openapi
 BOILERPLATE_FILENAME := vendor/k8s.io/code-generator/hack/boilerplate.go.txt
 REPORT_FILENAME := $(OUT_DIR)/violations.report
 KNOWN_VIOLATION_FILENAME := api/api-rules/violation_exceptions.list
```

Comment:
Currently the workflow is: API authors see the error message below when `violations.report` differs from `violation_exceptions.list`, and they must follow the instruction in the message to either 
- fix new API rule violations that they introduced
- run `hack/update-api-violation-exceptions.sh` to update the exception list when necessary

Alternatively we could have `REPORT_FILENAME := api/api-rules/violation_exceptions.list`, by which we can get rid of the hack script-- the API authors won't see any error when there are new violations, the `violation_exceptions.list` will be updated, and it becomes the API reviewers' responsibility to catch the file change and ask API authors to fix the violations.

Reply:
/cc @lavalamp 
for high level workflow review
/cc @ixdy 
for Makefile and hack script review

Reply:
I'm fine with either option--whatever @ixdy thinks is best.

Reply:
> Just generating the exceptions list automatically seems like the best experience for contributors. Do we expect api-approvers to notice changes to this list?

I suspect that's unlikely to happen. When api-authors add/update APIs and send a PR, I'd expect they put the auto-updated exceptions into a ["generated"](https://github.com/kubernetes/kubernetes/search?q=generated&type=Commits) commit, along with the other auto-generated files (bazel BUILD, openapi/swagger specs, api-reference docs, deepcopy, etc.). I think those commits usually are not (and shouldn't be) well-reviewed.

It would be an annoying extra step for api-reviewers to search for the exceptions filename in the "generated" commits, especially when most of the time that file shouldn't be changed.

> Yet another option would be to optionally make Makefile.generated_files update the known violations file

I love this idea! Thanks @ixdy. Let me try implementing it

Reply:
@ixdy Updated. Please take a look